src/frontend/src/scheduler/distributed/query.rs

```diff
@@ -138,9 +138,13 @@ impl QueryExecution {
     }
 
     /// Start execution of this query.
+    /// Note the two shutdown channel sender and receivers are not dual.
+    /// One is used for propagate error to `QueryResultFetcher`, one is used for listening on
+    /// ctrl-c.
```

Comment:
It doesn't not have to come from `ctrl-c`, we may have risectl/ui to cancel queries.

Reply:
Sure. This design is handle for all. I'll fix the doc